### ISSUE TITLE
Extremely High Background Battery Usage 

### ISSUE THREAD

**Issue by grravity** [Telegram-FOSS-Team/Telegram-FOSS#486]
Getting extremely high battery usage, 4min screen on time resulted in 33% of battery loss. Before charging I had about an 45min screen on time and it was 90%.

My second highest app is only around 4% usage.

I would say it is due to websockets, however I don't entirely think that's the case. Especially because none of my other apps are anywhere near as high.
Unless it's due to being in a large amount of groups or channels?

**Comment by c33s** [comment]
high battery usage is most of the time high cpu usage. maybe related to #390

**Comment by grravity** [comment]
Could possibly be, I don't believe I have a built in cpu measurment on my skin, but it is showing Telegram uses 1.4gb of memory which seems to be quite a bit.

**Comment by Heretical-Anonymous** [comment]
Same shit
![Screenshot_20210115-153252_CCleaner](https://user-images.githubusercontent.com/77483843/104701042-fe251800-570c-11eb-978a-b0a054647233.jpg)

**Comment by Heretical-Anonymous** [comment]
I'm deleting this for now because it's used almost 50% of battery in 2.5 hours its crazy

**Comment by c33s** [comment]
if i notice that my phone gets warm, my workaroud  is that i force-kill telegram mulitple times (it is always telegram which heats up my phone and drains the battery with a cpu usage of up to 60%). if that does not lead to lesser cpu usage i disable the background connection (settings > notification and sounds > other > background connection) for an hour or a day, after that the high cpu usage is always gone

**Comment by grravity** [comment]
Yeah this is crazy, this has definitely been only since one of the last 2 or 3 updates. I've had telegram-foss installed for over a year no issues.
Not sure if it was telegram or just telegram foss, bit the battery drain is kiling my phone, and I need to keep telegram on it.

Any official word as to whether or not this is getting worked on?

**Comment by zk-pre** [comment]
I also have Telegram FOSS drain 38% of battery, followed by Discord which is 6% but I had longer open today

**Comment by zk-pre** [comment]
hours longer in fact

**Comment by zk-pre** [comment]
My theory is that Telegram fetches messages and media in the background, including from those chats that are muted and wouldn't produce a notification anyway. And with more active groups and channels in the more battery telegram drains in the background.

Could the push notification service only listen to chats that aren't muted?

**Comment by grravity** [comment]
This was never the issue for me until one of the recent updates, however I +1 to that if it's not already a thing.

**Comment by mortezadadgar** [comment]
![Screenshot_20210223-210203_Termux](https://user-images.githubusercontent.com/27911493/108883692-c3e44b80-75d3-11eb-8473-4a920124e7a5.png)
I just spotted high CPU usage from app, is there any logs I can provide to help mitigate this issue?

**Comment by grravity** [comment]
Unrelated but was just curious how you got top working in tmux for android processes? Is it a root only thing? @mortezadadgar

**Comment by mortezadadgar** [comment]
> Unrelated but was just curious how you got top working in tmux for android processes? Is it a root only thing? @mortezadadgar

Yes "top" requires root to show all processes

**Comment by Fincer** [comment]
By default, the application seems to wake up a phone in 1 minute intervals if `pushService` is enabled, according to [org.telegram.messenger.ApplicationLoader.java, line 245](https://github.com/Telegram-FOSS-Team/Telegram-FOSS/blob/master/TMessagesProj/src/main/java/org/telegram/messenger/ApplicationLoader.java#L245).

```
am.setRepeating(AlarmManager.RTC_WAKEUP, System.currentTimeMillis(), 60000, pendingIntent);
```

I self-compiled the application after commenting out this line, and set `Repeat Notifications` to 5 minutes in in-app Settings menu (Notifications and Sounds). The first results show that application wake up alarms count dropped *dractically* (from dozens to a few) and are in line with the alarm count of Whatsapp. I haven't done any long-term testing for Telegram FOSS with this modification applied so I can't tell if the application functions such as receiving messages still work as intended, however.

**Comment by vomver** [comment]
Same here. The battery drains too fast and CPU usage is crazy. I disabled background notifications in preferences, and now works.

**Comment by lucker999** [comment]
+1
Every minute wake up is a bit  of extreme. 75% of battery usage is Telegram FOSS. This is unacceptable!

Please reduce the wake up calls at last down to once  in 5 minutes,  as for example WhatsApp does

**Comment by franzcor** [comment]
Having the same issue here. I had to disable background battery usage. @thermatk can you look into this? Looks like @Fincer somehow fixed the issue..I'd like to test it

**Comment by proletarius101** [comment]
> Guys, it's offtopic here, let's move out. For the whole Telegram-FOSS case I personally think that adding UnifiedPush would generally solve this issue. And IMHO it's currently the only patch that would help in the near future.

I won't oppose more mature solution. But before anyone working on that big project, I just want to propose simple mitigations that could make telegram Foss meaningful for daily usage

**Comment by proletarius101** [comment]
@luvletter2333 It's merged. Are you interested in pushing it to https://github.com/NekoX-Dev/NekoX?

**Comment by luvletter2333** [comment]
> @luvletter2333 It's merged. Are you interested in pushing it to https://github.com/NekoX-Dev/NekoX?

I have cherry-pick these two commits(#545 #543) from tfoss and made some changes. see [link](https://github.com/NekoX-Dev/NekoX/commit/a8b667084e13ab8992c79dca4ae39c39b7f81e29#diff-da13bd8469fe2083f56d4f404375fb08a88a3daf8b726e87fff22bd75f1dcc4aR400).

Besides, I distinguish pushConnection from pushService and store them in user-preferences instead of GlobalNotificationPreference. Hopefully these changes may solve some problems.

Thanks for your contributions.

**Comment by BigmenPixel0** [comment]
![IMG_20220824_192058](https://user-images.githubusercontent.com/73637218/186443135-f834f0cb-771d-42a2-aaee-5981e78a2322.jpg)
And high RAM usage...

**Comment by daitj** [comment]
Reporting similar issue like @dbrgn, I recently upgraded to Android 13 (LineageOS with microg), first I thought it was the update but after disabling background notification service in Telegram-FOSS the drain was not significant.  
I guess something is going on with background notification service server lately? 
I have Signal installed on the side as well, which uses similar tech to fetch notification when installed in devices without Google services, it doesn't have the battery drain issue. 
Non FOSS version uses Firebase so I guess the issue doesn't exist there.

**Comment by StrikS** [comment]
Android 11 (LineageOS with microg) same problem. ~20% battery consumption, but I almost never used Telegram, it only worked in the background.

**Comment by kostadinstoilov** [comment]
I've used Telegram FOSS for 6 years and only recently started observing high battery background usage. It happens on both of my Android 13 devices.

The high usage goes away if I Force Stop the app and restart it, but it comes back after some time.

I am yet to figure out what triggers it.

**Comment by Eloxan** [comment]
Same here. GrapheneOS an Telegram FOSS high Battery consumption since a few weeks.

**Comment by SigHunter** [comment]
This currently happens to me too. Telegram FOSS 9.3.3. My phone is a Oneplus 8T and it happened on OxygenOS Android 13 and I recently switched to LineageOS 20 and it still happens there. About 45 % of a full battery load (100 % to 0 %) is used by telegram. About 3 % per hour for telegram just when the phone is lying on the table with screen off. 
I experimented with the android app battery settings and it happens not only when set on "unrestricted" but also when set to "optimized", which is the default and tg normally it needs "unrestricted" (to my knowledge)

**Comment by RodoMa92** [comment]
Same issue on a POCO F3 running CRDroid Android 13 with microG. Battery dead in less than half a day. Never had any issue with battery consumption in background before with this setup. Killing the process manually fixes it temporarily but the CPU hog will come back in a few hours, usually. I always see 20/50% of the CPU utilized from telegram.

**Comment by RodoMa92** [comment]
It seems to be gone with the latest release, I'll keep monitoring just in case.

**Comment by SigHunter** [comment]
It is gone for me too, nice, thanks for the update

**Comment by prosenboim** [comment]
Started happening for me after latest updates. Google Pixel 8
